tests/test.py

```diff
@@ -9400,6 +9400,54 @@ def testClientIDRead(self):
             self.assertIsInstance(value, bytearray, column)
             self.assertNotEqual(data[column], value, column)
 
+class TestPostgresqlConnectWithTLSPrefer(BaseTestCase):
+    def checkSkip(self):
+        if TEST_WITH_TLS or not TEST_POSTGRESQL:
+            self.skipTest("running tests with TLS")
+
+    def with_tls(self):
+        return False
+
+    def fork_acra(self,  popen_kwargs: dict=None, **acra_kwargs: dict):
+        kwargs = {
+            'client_id': TLS_CERT_CLIENT_ID_1,
+            'tls_client_id_from_cert': False,
+        }
+        acra_kwargs.update(kwargs)
+        return super().fork_acra(popen_kwargs, **acra_kwargs)
+
+    def setUp(self):
+        self.checkSkip()
+        try:
+            if not self.EXTERNAL_ACRA:
+                self.acra = self.fork_acra()
+
+        except:
+            self.tearDown()
+            raise
+
+    def testPlainConnectionAfterDeny(self):
```

Comment:
I suggest to rewrite this function:
```
def testPlainConnectionAfterDeny():
    # declare fully async function to call run_until_complete once. better to separate async and sync code
    async def _testPlainConnectionAfterDeny(self):
      conn = await async.connect(...)
      conn.fetch(...)

    loop = asyncio.new_event_loop()  # create new to avoid concurrent usage of the loop in the current thread and allow parallel execution in the future
    loop.run_until_complete(_testPlainConnectionAfterDeny())

```